From 85c33f6a5b410ec8cabd81f211a0bfe11dedbf32 Mon Sep 17 00:00:00 2001
From: dann frazier <dann.frazier@chainguard.dev>
Date: Tue, 30 Dec 2025 15:25:18 -0700
Subject: [PATCH] Disable vendoring of files from other APKs

Upstream's build copies files out of a Docker container and vendors
them into this backend's private directory. We depend on the packages
that provide those files instead so that we always use the latest patched
versions.

Signed-off-by: dann frazier <dann.frazier@chainguard.dev>
---
 CMakeLists.txt | 34 +---------------------------------
 1 file changed, 1 insertion(+), 33 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b0e399..83027e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,38 +152,6 @@ endif() # TRITON_ENABLE_NVTX
 #
 configure_file(src/libtriton_pytorch.ldscript libtriton_pytorch.ldscript COPYONLY)
 
-set(PT_LIBS
-    "libc10.so"
-    "libc10_cuda.so"
-    "libtorch.so"
-    "libtorch_cpu.so"
-    "libtorch_cuda.so"
-    "libtorch_cuda_linalg.so"
-    "libtorch_global_deps.so"
-    "libjpeg.so.62"
-)
-
-if (${TRITON_PYTORCH_NVSHMEM})
-  set(PT_LIBS
-      ${PT_LIBS}
-      "libtorch_nvshmem.so"
-  )
-endif() # TRITON_PYTORCH_NVSHMEM
-
-if (${TRITON_PYTORCH_ENABLE_TORCHVISION})
-  set(PT_LIBS
-      ${PT_LIBS}
-      $<IF:$<BOOL:${RHEL_BUILD}>,libtorchvision.so,libtorchvision.so.1>
-  )
-endif() # TRITON_PYTORCH_ENABLE_TORCHVISION
-
-if (${TRITON_PYTORCH_ENABLE_TORCHTRT})
-  set(PT_LIBS
-      ${PT_LIBS}
-      "libtorchtrt_runtime.so"
-  )
-endif() # TRITON_PYTORCH_ENABLE_TORCHTRT
-
 if (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "aarch64")
   set(LIBS_ARCH "aarch64")
   set(LIBTORCH_LIBS
@@ -375,7 +343,7 @@ if (${TRITON_PYTORCH_DOCKER_BUILD})
   if (${TRITON_PYTORCH_ENABLE_TORCHVISION})
     set(TRITON_PYTORCH_LIBS
     ${TRITON_PYTORCH_LIBS}
-    "${CMAKE_CURRENT_BINARY_DIR}/$<IF:$<BOOL:${RHEL_BUILD}>,libtorchvision.so,libtorchvision.so.1>")
+    "${CMAKE_CURRENT_BINARY_DIR}/libtorchvision.so")
   endif() # TRITON_PYTORCH_ENABLE_TORCHVISION
 
   if (${TRITON_PYTORCH_ENABLE_TORCHTRT})
-- 
2.51.0

